### PR TITLE
ARROW-13401 [R]: Update python.Rmd vignette

### PR DESCRIPTION
**Pull request by AlenkaF**

### CODE REVIEW

File: r/vignettes/python.Rmd [hunk 1]
```diff
@@ -45,7 +55,7 @@ use_virtualenv("arrow-env")
 pa <- import("pyarrow")
 ```
 
-The package includes support for sharing Arrow `Array` and `RecordBatch`
+The `pyarrow` package includes support for sharing Arrow `Array` and `RecordBatch`
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:58]:
```suggestion
The pyarrow package includes support for sharing Arrow `Array` and `RecordBatch`
```

File: r/vignettes/python.Rmd [hunk 2]
```diff
@@ -45,7 +55,7 @@ use_virtualenv("arrow-env")
...
 objects in-process between R and Python. For example, let's create an `Array`
 in `pyarrow`.
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:60]:
```suggestion
in pyarrow.
```

File: r/vignettes/python.Rmd [hunk 3]
```diff
@@ -62,8 +72,8 @@ a
 ## ]
 ```
 
-`a` is now an `Array` object in our R session, even though we created it in Python.
-We can apply R methods on it:
+`a` is now an `Array` object in our R session, even though you created it in Python.
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:75]:
```suggestion
`a` is now an `Array` object in your R session, even though you created it in Python.
```

File: r/vignettes/python.Rmd [hunk 4]
```diff
@@ -76,10 +86,10 @@ a[a > 1]
 ## ]
 ```
 
-We can send data both ways. One reason we might want to use `pyarrow` in R is
+You can send data both ways. One reason you might want to use `pyarrow` in R is
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:89]:
```suggestion
You can send data both ways. One reason you might want to use pyarrow in R is
```

File: r/vignettes/python.Rmd [hunk 5]
```diff
@@ -76,10 +86,10 @@ a[a > 1]
...
 to take advantage of functionality that is better supported in Python than in R.
 For example, `pyarrow` has a `concat_arrays` function, but as of 0.17, this
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:91]:
```suggestion
For example, pyarrow has a `concat_arrays()` function, but as of 0.17, this
```

File: r/vignettes/python.Rmd [hunk 6]
```diff
@@ -76,10 +86,10 @@ a[a > 1]
...
-function is not implemented in the `arrow` R package. We can use `reticulate`
+function is not implemented in the `arrow` R package. You can use `reticulate`
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:92]:
```suggestion
function is not implemented in the arrow R package. You can use reticulate
```

File: r/vignettes/python.Rmd [hunk 7]
```diff
@@ -101,13 +111,24 @@ a_and_b
 ## ]
 ```
 
-Now we have a single `Array` in R.
+Now you have a single `Array` in R.
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:114]:
```suggestion
Now you have a single Array in R.
```

File: r/vignettes/python.Rmd [hunk 8]
```diff
@@ -101,13 +111,24 @@ a_and_b
...
 
 "Send", however, isn't the correct word. Internally, we're passing pointers to
 the data between the R and Python interpreters running together in the same
 process, without copying anything. Nothing is being sent: we're sharing and
 accessing the same internal Arrow memory buffers.
 
+## Arrow object types
+
+For more information about Arrow object types see the Internals section of
+the Arrow vignette by calling:
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:126]:
```suggestion
the "arrow" vignette:
```

File: r/vignettes/python.Rmd [hunk 9]
```diff
@@ -101,13 +111,24 @@ a_and_b
...
+## How this works
 
 "Send", however, isn't the correct word. Internally, we're passing pointers to
 the data between the R and Python interpreters running together in the same
 process, without copying anything. Nothing is being sent: we're sharing and
 accessing the same internal Arrow memory buffers.
 
+## Arrow object types
+
+For more information about Arrow object types see the Internals section of
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:125]:
```suggestion
For more information about Arrow object types see the "Internals" section of
```

File: r/vignettes/python.Rmd [hunk 10]
```diff
@@ -10,6 +10,11 @@ vignette: >
 The `arrow` package provides `reticulate` methods for passing data between
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:10]:
```suggestion
The arrow package provides [reticulate](https://rstudio.github.io/reticulate/) methods for passing data between
```

**Review by thisisnic — COMMENTED** [review]
This is looking good!  Most of the changes I've suggested are stylistic things we updated across the other R documentation which I forgot to mention in this ticket.  Your explanations of things here look great - thanks for adding in those bits!

File: r/vignettes/python.Rmd [hunk 11]
```diff
@@ -45,9 +54,9 @@ use_virtualenv("arrow-env")
 pa <- import("pyarrow")
 ```
 
-The package includes support for sharing Arrow `Array` and `RecordBatch`
+The pyarrow package includes support for sharing Arrow `Array` and `RecordBatch`
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:57]:
```suggestion
The arrow R package include support for sharing Arrow `Array` and `RecordBatch`
```

**Review by thisisnic — APPROVED** [review]
One tiny additional change, otherwise it's good to go

File: r/vignettes/python.Rmd (hunk 11 above)
Comment (AlenkaF) [reply r/vignettes/python.Rmd:57]: Heh, got that one wrong!

**Comment by AlenkaF** [comment]
Info: committed the changes for this PR completely through the GitHub editor!

File: r/vignettes/python.Rmd [hunk 12]
```diff
@@ -7,9 +7,13 @@ vignette: >
   %\VignetteEncoding{UTF-8}
 ---
 
-The `arrow` package provides `reticulate` methods for passing data between
+The arrow package provides [reticulate](https://rstudio.github.io/reticulate/) methods for passing data between
 R and Python in the same process. This document provides a brief overview.
 
+Why you might want to use `pyarrow`?
+* To use some Python functionality that is not yet implemented in R, for example, the `concat_arrays` function.
+* To transfer Python objects into R, for example, a Pandas dataframe into an R Arrow Array.
```
Comment (thisisnic) [inline r/vignettes/python.Rmd:15]:
```suggestion
Why you might want to use `pyarrow`?

* To use some Python functionality that is not yet implemented in R, for example, the `concat_arrays` function.
* To transfer Python objects into R, for example, a Pandas dataframe into an R Arrow Array. 
```
Needs a space to render properly

Comment (AlenkaF) [reply r/vignettes/python.Rmd:15]: Ok, I see I need to build the docs next time, sorry for giving you additional work! I wanted to try how it goes ...

**Pull request closed without merging**